test/build_processed_data_test.py

```diff
@@ -1,6 +1,57 @@
-from libs import build_processed_dataset
+from unittest.mock import patch
 
+import pytest
 
-def test_get_testing_df():
-    testing_df = build_processed_dataset.get_testing_timeseries_by_state("MA")
-    assert testing_df is not None
+from libs import build_processed_dataset, validate_results
+from libs.datasets import CommonFields, CDSDataset, CovidTrackingDataSource
+import pandas as pd
+from libs.datasets.results_schema import CUMULATIVE_POSITIVE_TESTS, CUMULATIVE_NEGATIVE_TESTS
+
+from libs.enums import Intervention
+from more_itertools import one
+
+
+def test_get_usa_by_states_df():
+    empty_df_with_state = pd.DataFrame([], columns=["state",])
+    with patch(
```

Comment:
this is totally fine and no need to change, just wanted to enumerate a couple other pytest fixture options:
 - monkeypatch fixture that pytest has [included](https://docs.pytest.org/en/latest/monkeypatch.html)
 - i've liked [pytest-mock](https://pypi.org/project/pytest-mock/)

an di

Reply:
Thank you for the pointers. I find fixtures a bit voodoo but could get into `with monkeypatch.context() as m:`